src/main/java/hvac/roomcoordinator/RoomCoordinatorAgent.java

```diff
@@ -0,0 +1,245 @@
+package hvac.roomcoordinator;
+
+import hvac.meetingontology.Meeting;
+import hvac.meetingontology.MeetingOntology;
+import hvac.meetingontology.Request;
+import hvac.meetingontology.RequestStatus;
+import hvac.simulation.rooms.RoomWall;
+import jade.content.lang.Codec;
+import jade.content.lang.sl.SLCodec;
+import jade.content.onto.Ontology;
+import jade.core.AID;
+import jade.core.Agent;
+import jade.core.behaviours.CyclicBehaviour;
+import jade.core.behaviours.TickerBehaviour;
+import jade.lang.acl.ACLMessage;
+
+import java.util.Date;
+import java.util.HashMap;
+
+@SuppressWarnings("unused")
+public class RoomCoordinatorAgent extends Agent {
+    private RoomContext roomContext;
+    private TickerBehaviour tickerBehaviour;
+    private static Codec codec = new SLCodec();
+    private static Ontology meetingOntology = MeetingOntology.getInstance();
+
+    @Override
+    protected void setup() {
+        getContentManager().registerLanguage(codec);
+        getContentManager().registerOntology(meetingOntology);
+        Object[] args =  getArguments();
+        HashMap<AID, RoomWall> roomConfig = null;
+        try {
+            @SuppressWarnings("unchecked")
+            HashMap<AID, RoomWall> hashMap = (HashMap<AID, RoomWall>) args[2];
+            hashMap.keySet().forEach(x -> {});
+            hashMap.values().forEach(x -> {});
```

Comment:
Może dasz jakiś komentarz z jakim argumentem to uruchamiać i co robią te .forEach(x -> {})?

Reply:
Spróbuję to jakoś opisać, ale sam w sumie nie wiem dokładnie dlaczego muszę tak robić - wiem że powinienem, ale nie znam na tyle Javy zeby wiedziec dlaczego mapowanie takich obiektów nie jest łatwiejsze...